scripts/preprocess_cloudsat/data_preprocessing.py

```diff
@@ -0,0 +1,292 @@
+        'longitude',
+        'height',
+        'iwc',
+        'iwp',
+        'time_since_start',
+    ]
+
+    latlim = (-70, 69)  #  latitude limits for Gridsat
+
+    ds = l2c_ice.open(filepath)
+    ds = ds.reset_coords()
+    # keep selected variables
+    ds = ds[variables]
+    # filter on latitudes
+    idxs = ds.latitude > latlim[0]
+    ds = ds.isel(rays=idxs)
+    idxs = ds.latitude < latlim[1]
+    ds = ds.isel(rays=idxs)
+    # reverse 'bins' coordinate to make height go from low to high values
+    ds = ds.reindex(bins=ds.bins[::-1])
+
+    return ds
+
+def load_cldclass(filepath : Path):
+
+    # 'rays' and 'bins' are dimensions
+    variables = [
+        'latitude',
+        'height',
+        'surface_elevation',
+        'cloud_class',
+        'cloud_class_flag',
+    ]
+
+    latlim = (-70, 69)  #  latitude limits for Gridsat
+
+    ds = l2b_cldclass.open(filepath)
+    ds = ds.reset_coords()
+    # keep selected variables
+    ds = ds[variables]
+    # filter on latitudes
+    idxs = ds.latitude > latlim[0]
+    ds = ds.isel(rays=idxs)
+    idxs = ds.latitude < latlim[1]
+    ds = ds.isel(rays=idxs)
+    # reverse 'bins' coordinate to make height go from low to high values
+    ds = ds.reindex(bins=ds.bins[::-1])
+
+    return ds
+
+def load_ccic(filepath : Path):
+
+    variables = [
+        'cloud_prob_2d',
+        'tiwp',
+    ]
+    with xr.open_dataset(filepath, engine='zarr') as ds:
+        # Filter on latitude
+        # NOTE latitudes in the cpcir dataset goes from 60 to -60 deg
+        # ds = ds.sel(
+        #     latitude=slice(latlim[1], latlim[0]),
+        # )
+        # # Split dataset into separate utc times
+        # data = [ds.isel(time=0), ds.isel(time=1)]
+        # keep selected variables
+        ds = ds[variables]
+        return ds
+    
+def load_ccic_grid(local_time_grid, ccic_datadir):
+    """
+    """
+    dsets = []
+    for t in local_time_grid:
+        timestamp = t.strftime("%Y%m%d%H%M")
+        file_to_load = f'{timestamp[0:4]}/ccic_gridsat_{timestamp}.zarr'
+        filepath = Path(ccic_datadir, file_to_load)
+        # Load data
+        dsets.append(load_ccic(filepath))
+    return xr.concat(dsets, dim='time')
+
+def load_era5(filepath : Path):
+
+    variables = [
+        'tiwp',
+        'tcc',
+    ]
+    with xr.open_dataset(filepath) as ds:
+        # add column snow and column ice to get total ice water path
+        ds['tiwp'] = ds.tcsw + ds.tciw
+        ds = ds[variables]
+        return ds
+
+def load_era5_grid(local_time_grid, datadir):
+    """
+    """
+    dsets = []
+    for t in local_time_grid:
+        timestamp = t.strftime("%Y%m%d%H")
+        file_to_load = f'reanalysis-era5-single-levels_{timestamp}_total_column_snow_water-total_cloud_cover-total_column_cloud_ice_water-90-90--180-180.nc'
+        filepath = Path(datadir, file_to_load)
+        # Load data
+        dsets.append(load_era5(filepath))
+    return xr.concat(dsets, dim='time')
```

Comment:
Does this correspond to this PR?
I would suggest to remove it and place it in a different branch, to restrict this PR to cloudsat data and GridSat collocations.